### ISSUE TITLE
multibinder-haproxy-wrapper occasionally dies after not finding haproxy PID file

### ISSUE THREAD

**Issue by westse** [github/multibinder#13]
Using [haproxy-ingress](https://github.com/jcmoraisjr/haproxy-ingress) and occasionally see the following error which kills the multibinder process

```
/usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:42:in `read': No such file or directory @ rb_sysopen - /var/run/haproxy.pid (Errno::ENOENT)
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:42:in `block (2 levels) in <top (required)>'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:41:in `each'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:41:in `block in <top (required)>'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:60:in `sleep'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:60:in `block in <top (required)>'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:58:in `loop'
	from /usr/lib/ruby/gems/2.3.0/gems/multibinder-0.0.4/bin/multibinder-haproxy-wrapper:58:in `<top (required)>'
	from /usr/bin/multibinder-haproxy-wrapper:22:in `load'
	from /usr/bin/multibinder-haproxy-wrapper:22:in `<main>'
```

**Comment by theojulienne** [comment]
Thanks for the report! Looks like [haproxy unlinks the pid file momentarily](https://github.com/haproxy/haproxy/blob/master/src/haproxy.c#L1902-L1903) so there's a possibility of a race condition here. Can you see if https://github.com/github/multibinder/pull/14 solves the issue for you? I think that should protect against this case.

**Comment by westse** [comment]
@theojulienne, thanks for looking at this. How do you feel about producing a release that includes this fix? The fix looks pretty safe, and that would be the easiest way for us to test it since we use multibinder indirectly through our usage of [haproxy-ingress](https://github.com/jcmoraisjr/haproxy-ingress) (which consumes the multibinder gem).

**Comment by theojulienne** [comment]
> How do you feel about producing a release that includes this fix? The fix looks pretty safe

:+1: We actually happened to test a similar code change elsewhere, so I've released it as `0.0.5`.

**Comment by westse** [comment]
Looks like that did it. Thanks!
